plugins/gpg-agent/gpg-agent.plugin.zsh

```diff
@@ -1,14 +1,17 @@
-# Enable gpg-agent if it is not running
-GPG_AGENT_SOCKET="${XDG_RUNTIME_DIR}/gnupg/S.gpg-agent.ssh"
-if [ ! -S $GPG_AGENT_SOCKET ]; then
-  gpg-agent --daemon >/dev/null 2>&1
-  export GPG_TTY=$(tty)
+# Enable gpg-agent if it is not running-
+# --use-standard-socket will work from version 2 upwards
+
+AGENT_SOCK=`gpgconf --list-dirs | grep agent-socket | cut -d : -f 2`
```

Comment:
GPG_AGENT_SOCKET=`gpgconf --list-dirs agent-ssh-socket`
would be enough like in my PR from two weeks ago (#6431 )

Reply:
I found that this didn't work in all cases, for example on an ubuntu 14.04 box gpg is on version  2.0.22 - On this box:  

gpgconf --list-dirs agent-ssh-socket

sysconfdir:/etc/gnupg2
bindir:/usr/bin
libexecdir:/usr/lib/gnupg2
libdir:/usr/lib/x86_64-linux-gnu/gnupg2
datadir:/usr/share/gnupg2
localedir:/usr/share/locale
dirmngr-socket:/var/run/dirmngr/socket
agent-socket:/home/chrisfl/.gnupg/S.gpg-agent
homedir:/home/chrisfl/.gnupg

Hence the longer version.

Reply:
@chrisfleming Okay, so why won't you use agent-socket and add it? Check if GPG>=2.1.13 then use `agent-ssh-socket`, if not then use `agent-socket` and eventually add .ssh to it?

Reply:
Description:	Ubuntu 16.04.3 LTS
Release:	16.04
Codename:	xenial

```
gpg --version
gpg (GnuPG) 2.1.11
libgcrypt 1.6.5
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Home: ~/.gnupg
Supported algorithms:
Pubkey: RSA, ELG, DSA, ECDH, ECDSA, EDDSA
Cipher: IDEA, 3DES, CAST5, BLOWFISH, AES, AES192, AES256, TWOFISH,
        CAMELLIA128, CAMELLIA192, CAMELLIA256
Hash: SHA1, RIPEMD160, SHA256, SHA384, SHA512, SHA224
Compression: Uncompressed, ZIP, ZLIB, BZIP2
```

```
gpgconf --list-dirs agent-ssh-socket
sysconfdir:/etc/gnupg2
bindir:/usr/bin
libexecdir:/usr/lib/gnupg2
libdir:/usr/lib/x86_64-linux-gnu/gnupg2
datadir:/usr/share/gnupg2
localedir:/usr/share/locale
dirmngr-socket:/home/bram/.gnupg/S.dirmngr
dirmngr-sys-socket:/var/run/gnupg2/S.dirmngr
agent-socket:/home/bram/.gnupg/S.gpg-agent
homedir:/home/bram/.gnupg
```